### PR TITLE
Top 5 merchants by revenue jr

### PR DESCRIPTION
**Pull request by jimriddle1**
Hey everyone, 

This PR will satisfy the user story for top 5 merchants by total revenue.  We are getting some incomplete test coverage on the merchant items controller, but I'm not too worried about it, we still have plenty of time.

Let me know if you have any questions.  This includes feature and model testing.

-Jim

#11

### CODE REVIEW

**Comment by jimriddle1** [comment]
looks like there were some conflicts on the model / merchant spec.

File: app/controllers/admin_merchants_controller.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-class AdminMerchantsController < ApplicationController 
+class AdminMerchantsController < ApplicationController
     def index
         @merchants = Merchant.all
     end
-    
-end
 
+    def show
+      @merchant = Merchant.find(params[:id])
```
Comment (z-prince) [inline app/controllers/admin_merchants_controller.rb:8]: This looks superb!

File: app/models/item.rb [hunk 2]
```diff
@@ -20,12 +20,12 @@ def find_invoice_id
   def item_best_day
     invoices.joins(:transactions)
             .where(transactions: {result: 0})
-            .group(:id)
-            .select("invoices.*, sum(invoice_items.quantity) as revenue")
-            .order(revenue: :desc)
```
Comment (z-prince) [inline app/models/item.rb:25]: Good job with this advanced AR!

File: app/models/merchant.rb [hunk 3]
```diff
@@ -16,10 +16,26 @@ def ready_items
   def top_five_items_by_revenue
     items.joins(invoice_items: :transactions)
          .where(transactions: {result: 0})
-         .group(:id)
```
Comment (z-prince) [inline app/models/merchant.rb:19]: You're killing it!

File: app/models/merchant.rb [hunk 4]
```diff
@@ -16,10 +16,26 @@ def ready_items
...
+         .group(:id) #item_id
          .select("items.*, sum(quantity * invoice_items.unit_price) as total_revenue")
          .order(total_revenue: :desc)
          .limit(5)
   end
-  
+
+  def self.top_five_merchants_by_revenue
```
Comment (z-prince) [inline app/models/merchant.rb:25]: This is a thing of beauty.

File: config/routes.rb [hunk 5]
```diff
@@ -2,6 +2,7 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   get '/admin', to: 'admins#dashboard'
```
Comment (z-prince) [inline config/routes.rb:3]: Would love to refactor these to advanced routes!

File: spec/features/admin/merchants/index_spec.rb [hunk 6]
```diff
@@ -11,11 +11,68 @@
...
+            @invoice3.transactions.create!(result: 1)
+            InvoiceItem.create!(item_id: @item1.id, invoice_id: @invoice1.id, quantity: 1, unit_price: 10000, status: 0)
+            InvoiceItem.create!(item_id: @item2.id, invoice_id: @invoice2.id, quantity: 2, unit_price: 10000, status: 1)
+            InvoiceItem.create!(item_id: @item3.id, invoice_id: @invoice1.id, quantity: 3, unit_price: 10000, status: 1)
+            InvoiceItem.create!(item_id: @item4.id, invoice_id: @invoice2.id, quantity: 4, unit_price: 10000, status: 2)
+            InvoiceItem.create!(item_id: @item5.id, invoice_id: @invoice1.id, quantity: 5, unit_price: 10000, status: 0)
+            InvoiceItem.create!(item_id: @item6.id, invoice_id: @invoice2.id, quantity: 6, unit_price: 10000, status: 2)
+            InvoiceItem.create!(item_id: @item7.id, invoice_id: @invoice1.id, quantity: 1, unit_price: 500, status: 2)
+            InvoiceItem.create!(item_id: @item8.id, invoice_id: @invoice3.id, quantity: 1000, unit_price: 10000, status: 2)
+            InvoiceItem.create!(item_id: @item9.id, invoice_id: @invoice1.id, quantity: 10, unit_price: 20000, status: 2)
```
Comment (z-prince) [inline spec/features/admin/merchants/index_spec.rb:50]: Holy smokes this test is huge!

**Review by z-prince — COMMENTED** [review]
Great work Jim!

**Pull request merged**